src/templates/common/socialLinks/index.tsx

```diff
@@ -1,28 +1,117 @@
 export interface SocialLinksProps {
   path: string
   title: string
+  isNews?: boolean
+  description?: string
+  address?: {
+    additionalName?: string | null
+    addressLine1?: string
+    addressLine2?: string
+    administrativeArea?: string
+    countryCode?: string
+    dependentLocality?: string | null
+    familyName?: string | null
+    givenName?: string | null
+    langcode?: string
+    locality?: string
+    organization?: string | null
+    postalCode?: string | null
+    sortingCode?: string | null
+  }
+  dateObject?: {
+    duration: number
+    endTime: string
+    endValue: number
+    end_value: string
+    rrule: number
+    rrule_index: number
+    startTime: string
+    timezone: string
+    value: number
+  }
 }
 
-export const SocialLinks = ({ path, title }: SocialLinksProps): JSX.Element => {
+export const SocialLinks = ({
+  path,
+  title,
+  isNews = false,
+  description,
+  address,
+  dateObject,
+}: SocialLinksProps): JSX.Element => {
+  if (isNews) {
```

Comment:
this feels a little funky to me. seems like `address`, `description`, `dateObject` are actually the optional values here (versus adding a boolean flag), and the additional item renders if they are present. Something like:

```
{description && address && dateObject && (
    <li className="vads-u-margin-bottom--2p5">
          <a
            data-description={description}
            data-end={dateObject.endValue}
            data-location={`${address.addressLine1} ${address.locality}, ${address.administrativeArea}`}
            data-start={dateObject.value}
            data-subject={title}
            href={path}
            id="add-to-calendar-link"
          >
            <i
              className="va-c-social-icon fas fa-calendar-check vads-u-margin-right--0p5"
              aria-hidden="true"
              role="presentation"
            ></i>
            Add to Calendar
          </a>
        </li>)}
 ```

Reply:
I went this route as it felt easier to deal with the different styling and classes used for the news vs non-news version.  Do you think it would be better to make them just be different components all together?

Reply:
looking at content-build, there are at least 4 different social share templates :sob:

I think let's make them different components for now, and see if we can streamline that logic in the future for next-build